pkg/protocol/engine/ledger/ledger/ledger.go

```diff
@@ -74,8 +76,20 @@ func NewProvider() module.Provider[*engine.Engine, ledger.Ledger] {
 			// TODO: how do we want to handle changing API here?
 			iotagoAPI := l.apiProvider.CurrentAPI()
 			l.manaManager = mana.NewManager(iotagoAPI.ManaDecayProvider(), iotagoAPI.ProtocolParameters().RentStructure(), l.resolveAccountOutput)
-			l.accountsLedger.SetCommitmentEvictionAge(iotagoAPI.ProtocolParameters().MaxCommittableAge())
-			l.accountsLedger.SetLatestCommittedSlot(e.Storage.Settings().LatestCommitment().Index())
+			maxCommittableAge := iotagoAPI.ProtocolParameters().MaxCommittableAge()
+			latestCommittedSlot := e.Storage.Settings().LatestCommitment().Index()
+			l.accountsLedger.SetCommitmentEvictionAge(maxCommittableAge)
```

Comment:
This should also not hardcode the value but fetch it from the active protocol parameters using an APIProvider